### PR TITLE
Fix loggers and db initialization when using multiple Gunicorn workers

### PR DESCRIPTION
**Pull request by armanddidierjean**
### Description

- disable default Gunicorn logger handlers to use our custom loggers
- remove `--preload` flag which break our async Queue logger handlers
- allow to prevent db init on Hyperion startup using an env variable
- add a Gunicorn on_starting hook to init the db

### Checklist

- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [x] Extended the documentation, if necessary

### CODE REVIEW

File: app/core/config.py [hunk 1]
```diff
@@ -157,10 +150,29 @@ class Settings(BaseSettings):
...
 
+    # If Hyperion should initialize the database on startup
+    # This environment variable is set by the Gunicorn on_starting hook, to tell the workers to avoid initializing the database
+    # You don't want to set this variable manually
+    HYPERION_INIT_DB: bool = True
+
+    # An old variable used by clients to determine if they are compatible with Hyperion
+    # Using 0.0.1 as the minimal version, force old and broken clients to show a message asking to update
+    # TODO: remove this property when old clients are updated
+    MINIMAL_TITAN_VERSION: str = "0.0.1"  # deprecated, use MINIMAL_TITAN_VERSION_CODE
```
Comment (julien4215) [inline app/core/config.py:174]: Maybe it is time to remove this property ?

If we keep it, why did you put it in automatically generated parameters ?

File: app/core/log.py [hunk 2]
```diff
@@ -220,6 +220,18 @@ def get_config_dict(self, settings: Settings):
...
+                "gunicorn.access": {"handlers": []},
+                "gunicorn.error": {
+                    "handlers": [
+                        "file_errors",
+                        "matrix_errors",
+                        "console",
+                    ],
+                    "level": MINIMUM_LOG_LEVEL,
+                    "propagate": False,
+                },
```
Comment (julien4215) [inline app/core/log.py:234]: I think this should be moved in a different PR. Maybe in this one #452 ?

**Review by julien4215 — COMMENTED** [review]
Amazing PR ! I did not have time to test it yet but I'll do it later.

We should add gunicorn as a dev dependency.

File: README.md [hunk 3]
```diff
@@ -200,3 +200,13 @@ To enable the service:
    1. Go to [Google cloud, IAM and administration, Service account](https://console.cloud.google.com/iam-admin/serviceaccounts) and add a new Service Account with Messaging API capabilities.
    2. Choose _Manage keys_ and create a new JSON key.
    3. Rename the file `firebase.json` and add it at Hyperion root
+
+---
+
+## Running Hyperion with Gunicorn
+
+For production we encourage to run Hyperion with multiple Gunicorn workers. You can use our [docker image](./Dockerfile) and [docker-compose file](./docker-compose.yaml) files to run Hyperion with Gunicorn.
```
[the remark this answers is not in the document]
Comment (armanddidierjean) [reply README.md:208]: I think we can use Gunicorn with Uvicorn workers, but I'm not sure we use that in production.

What would you think of:
```suggestion
For production we encourage to run Hyperion with Gunicorn and multiple workers. You can use our [docker image](./Dockerfile) and [docker-compose file](./docker-compose.yaml) files to run Hyperion with Gunicorn.
```

File: app/core/config.py (hunk 1 above)
Comment (armanddidierjean) [reply app/core/config.py:174]: Good idea, I will make a second PR

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (julien4215) [reply README.md:208]: I am pretty sure that the image we use run multiple Uvicorn workers with Gunicorn so I would suggest
```suggestion
For production we encourage to use Gunicorn to run and manage multiple Uvicorn workers. You can use our [docker image](./Dockerfile) and [docker-compose file](./docker-compose.yaml) files to run Hyperion with Gunicorn.
```

Maybe we could simply add a link to the FastAPI documentation https://fastapi.tiangolo.com/deployment/server-workers/#gunicorn-with-uvicorn-workers

[the remark this answers is not in the document]
Comment (armanddidierjean) [reply README.md:208]: You're completely right, mb.
```suggestion
For production we encourage to use Gunicorn to run and manage multiple Uvicorn workers. You can use our [docker image](./Dockerfile) and [docker-compose file](./docker-compose.yaml) files to run Hyperion with Gunicorn. See [Gunicorn with Uvicorn](https://fastapi.tiangolo.com/deployment/server-workers/#gunicorn-with-uvicorn-workers) FastAPI documentation.
```

**Comment by armanddidierjean** [comment]
> Don't we need to add gunicorn as a dev dependency ?
If we continue to use uvicorn as development server, I would say we don't need to add gunicorn as a dependency.

File: gunicorn.conf.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import logging
+import os
+
+from gunicorn.arbiter import Arbiter
```
Comment (julien4215) [inline gunicorn.conf.py:4]: We use gunicorn here and it will not be recognized if we don't add it to dev requirements

Comment (armanddidierjean) [reply gunicorn.conf.py:4]: Is it really an issue? The CI doesn't complain, if we don't install and use Gunicorn this file won't be used.

Comment (julien4215) [reply gunicorn.conf.py:4]: The CI doesn't complain because ruff still didn't implement E0401 rule.

I think we should give to developers every modules we use in our code.

Comment (julien4215) [reply gunicorn.conf.py:4]: Actually do we need to import the Arbiter class ? I believe we do not need to type `on_starting` `server` argument

Comment (armanddidierjean) [reply gunicorn.conf.py:4]: We don't really, the objective was to type the method, but we don't use the `server` argument. I will just add a comment

File: pyproject.toml [hunk 5]
```diff
@@ -108,7 +108,9 @@ disallow_any_generics = false
 disallow_untyped_calls = false
 
 [[tool.mypy.overrides]]
-module = ["firebase_admin", "icalendar", "fitz"]
+# Tracking issues
+# Gunicorn : https://github.com/benoitc/gunicorn/issues/2833
+module = ["firebase_admin", "icalendar", "fitz", "gunicorn.arbiter"]
```
Comment (julien4215) [inline pyproject.toml:113]: I think we can remove this too

**Review by Rotheem — APPROVED** [review]

**Review by Malem38 — APPROVED** [review]
